### ISSUE TITLE
New federation deployment mechanism exits with an error for non-GCP clusters.

### ISSUE THREAD

**Issue by madhusudancs** [kubernetes/kubernetes#33886]
As described here - https://github.com/kubernetes/kubernetes/issues/31071#issuecomment-250933234

```shell
KUBERNETES_PROVIDER=aws federation/deploy/deploy.sh deploy_federation
```

exits because the provider isn't gce/gke. However, the deployment mechanism itself should work on all the providers. We shouldn't exit, but just branch out to kubeconfig check instead.

cc @owenhaynes @kubernetes/sig-cluster-federation

**Comment by madhusudancs** [comment]
Fix to this should also be cherry-picked into the next 1.4.x patch release.

**Comment by madhusudancs** [comment]
@owenhaynes This is fixed in v1.4.2 now. Could you please verify?
